app/portainer/home/EnvironmentList/KubeconfigButton.tsx

```diff
@@ -34,48 +82,138 @@ export function KubeconfigButton({ environments }: Props) {
       category: 'kubernetes',
     });
 
-    showKubeconfigModal(environments);
+    selection.clear();
+    setShowDialog(true);
   }
-}
 
-function isKubeconfigButtonVisible(environments: Environment[]) {
-  if (window.location.protocol !== 'https:') {
-    return false;
+  function handleClose() {
+    setShowDialog(false);
+  }
+
+  function handleDownload() {
+    confirmKubeconfigSelection();
+  }
+
+  function isKubeconfigButtonVisible(environments: Environment[]) {
+    if (window.location.protocol !== 'https:') {
+      return false;
+    }
+    return environments.some((env) => isKubernetesEnvironment(env.Type));
+  }
+
+  async function confirmKubeconfigSelection() {
+    if (selection.size === 0) {
+      notifications.warning('No environment was selected', '');
+      return;
+    }
+    try {
+      await kcService.downloadKubeconfigFile(Array.from(selection));
+      setShowDialog(false);
+    } catch (e) {
+      notifications.error('Failed downloading kubeconfig file', e as Error);
+    }
   }
-  return environments.some((env) => isKubernetesEnvironment(env.Type));
 }
 
-async function showKubeconfigModal(environments: Environment[]) {
-  const kubeEnvironments = environments.filter((env) =>
+export interface KubeconfigPormptProps {
+  kubeServiceExpiry: string | undefined;
+  selection: Set<number>;
+  envQueryParams: EnvironmentsQueryParams;
+}
+const storageKey = 'home_endpoints';
+
+export function KubeconfigPormpt({
```

Comment:
this is a big enough component to have its own file

Reply:
and fix `KubeconfigPormpt` to `KubeconfigPrompt`. suggestion, use something like https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

Reply:
It's really useful, thank you. Installed and updated